spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -3254,6 +3308,92 @@ def _handle_goto_uri_event(self, pos):
             self._last_hover_uri = uri
             return False
 
+    def line_range(self, position):
+        """
+        Get line range from position.
+        """
+        if position is None:
+            return None
+        if position >= self.document().characterCount():
+            return None
+        # Check if still on the line
+        cursor = self.textCursor()
+        cursor.setPosition(position)
+        line_range = (cursor.block().position(),
+                      cursor.block().position()
+                      + cursor.block().length() - 1)
+        return line_range
+
+    def strip_trailing_spaces(self):
+        """
+        Strip trailing spaces if needed.
+
+        Remove trailing whitespace on leaving a non-string line containing it.
+        Return the number of removed spaces.
+        """
+        # Update current position
+        current_position = self.textCursor().position()
+        last_position = self.last_position
+        self.last_position = current_position
+
+        if self.skip_rstrip:
+            return 0
+
+        line_range = self.line_range(last_position)
+        if line_range is None:
+            # Doesn't apply
+            return 0
+
+        def pos_in_line(pos):
+            """Check if pos is in last line."""
+            if pos is None:
+                return False
+            return line_range[0] <= pos <= line_range[1]
+
+        if pos_in_line(current_position):
+            # Check if still on the line
+            return 0
+
+        if self.strip_automatic_spaces_only:

```

Comment:
I missed a spot somehow, that's why your tests are failing.

```suggestion
        if not self.strip_trailing_spaces_on_modify:
```